operations/agent_monitor.go

```diff
@@ -0,0 +1,427 @@
+
+	go func() {
+		id := 0
+		for {
+			logIDSource <- id
+			id++
+		}
+	}()
+}
+
+func getLogID() int {
+	return <-logIDSource
+}
+
+// monitor represents the agent monitor, including arguments necessary for it to
+// operate.
+type monitor struct {
+	// Monitor args
+	certificatePath string
+	clientURL       string
+	clientPath      string
+	logPrefix       string
+	jasperPort      int
+	port            int
+
+	// Args to be forwarded to the agent
+	agentArgs []string
+
+	// Set during runtime
+	jasperClient jasper.RemoteClient
+}
+
+// retryArgs defines the policy for retrying requests.
+type retryArgs struct {
+	maxAttempts int
+	minDelay    time.Duration
+	maxDelay    time.Duration
+}
+
+const (
+	defaultMonitorPort        = defaultAgentStatusPort - 1
+	defaultJasperPort         = defaultMonitorPort - 1
+	defaultMaxRequestDelay    = 30 * time.Second
+	defaultMaxRequestAttempts = 5
+
+	monitorLoggerName = "evergreen.agent.monitor"
+)
+
+// agentMonitor starts the monitor that deploys the agent.
+func agentMonitor() cli.Command {
+	const (
+		certificatePathFlagName = "certificate"
+		clientURLFlagName       = "client_url"
+		clientPathFlagName      = "client_path"
+		logPrefixFlagName       = "log_prefix"
+		jasperPortFlagName      = "jasper_port"
+		portFlagName            = "port"
+	)
+
+	return cli.Command{
+		Name:  "monitor",
+		Usage: "start the monitor on a host",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  certificatePathFlagName,
+				Usage: "the path to the certificate used to authenticate the monitor to Jasper",
+			},
+			cli.StringFlag{
+				Name:  clientURLFlagName,
+				Usage: "the url to fetch the evergreen client from",
+			},
+			cli.StringFlag{
+				Name:  clientPathFlagName,
+				Usage: "the name of the agent's evergreen binary",
+			},
+			cli.StringFlag{
+				Name:  logPrefixFlagName,
+				Value: monitorLoggerName,
+				Usage: "the prefix for the monitor's log name",
+			},
+			cli.IntFlag{
+				Name:  jasperPortFlagName,
+				Value: defaultJasperPort,
+				Usage: "the port that is running the Jasper RPC service",
+			},
+			cli.IntFlag{
+				Name:  portFlagName,
+				Value: defaultAgentStatusPort - 1,
+				Usage: "the port that used by the monitor",
+			},
+		},
+		Before: mergeBeforeFuncs(
+			requireStringFlag(clientURLFlagName),
+			requireStringFlag(clientPathFlagName),
+			requireIntValueBetween(jasperPortFlagName, 0, 1<<16),
+			requireIntValueBetween(portFlagName, 0, 1<<16),
+			func(*cli.Context) error {
+				grip.SetName(monitorLoggerName)
+				return nil
+			},
+		),
+		Action: func(c *cli.Context) error {
+			m := &monitor{
+				certificatePath: c.String(certificatePathFlagName),
+				clientURL:       c.String(clientURLFlagName),
+				clientPath:      c.String(clientPathFlagName),
+				jasperPort:      c.Int(jasperPortFlagName),
+				port:            c.Int(portFlagName),
+				logPrefix:       c.String(logPrefixFlagName),
+			}
+
+			args, err := getAgentArgs(c, os.Args)
```

Comment:
You could `m.agentArgs, err := getAgentArgs(c, os.Args)` here and eliminate `m.agentArgs = args` below, since the function returns in the error case.